src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -0,0 +1,395 @@
+/*******************************************************************\
+
+Module: C Nondet Symbol Factory
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <set>
+#include <sstream>
+
+#include <util/arith_tools.h>
+#include <util/fresh_symbol.h>
+#include <util/std_types.h>
+#include <util/std_code.h>
+#include <util/std_expr.h>
+#include <util/message.h>
+#include <util/namespace.h>
+#include <util/pointer_offset_size.h>
+#include <util/prefix.h>
+
+#include <linking/zero_initializer.h>
+
+#include <ansi-c/c_types.h>
+#include <ansi-c/string_constant.h>
+
+#include <goto-programs/goto_functions.h>
+
+#include "c_nondet_symbol_factory.h"
+
+class symbol_factoryt:public messaget
+{
+  code_blockt &init_code;
+  std::set<irep_idt> recursion_set;
```

Comment:
This variable isn't used, so I've deleted it. It's used in the java code to cope with structs, which the C code doesn't do yet.